### DISCUSSION TITLE
best way to highlight onHover object on map?

### DISCUSSION THREAD

**Discussion by tamir-jether** [visgl/deck.gl#5634] [Q&A]
I want to make objects on the map highlighted when the user hovers on them. I was wondering what's the best way to do so when using react + deck.gl.

Right now I'm using react + redux, and in the state object, I'm saving the data of the onHover object (which have a unique ID). Every time I render my Layer, on the getFillColor function for example, I check if my current object ID is the same ID of the currently onHover object I have in my global state. If it is I give it a different color.

Sometimes it's slow since I re-color everything every time I hover on an object.

My question is this - is that the "best way" to handle highlighting, or is there a different better way to do so?

**Comment by AdriSolid** [comment]
You can use the following props to speed it up:
```javascript
pickable: true,
autoHighlight: true,
highlightColor: [0, 255, 0]
```

**Reply by tamir-jether** [reply]
thanks!

but i wonder if this feature is broken in any way.

when using it, its "flickering". i only see the highlighted color when moving the mouse. (when i leave the mouse on the object, nothing happens, only when i move it)

**Reply by AdriSolid** [reply]
That's not broken, `autoHighlight` logic is made to highlight features when the mouse hovers any feature. If you want to hold the highlighted color after the event, you will need to implement some other approach then.

**Reply by tamir-jether** [reply]
i see, thanks!

on a similar subject (dont want to open a new thread).
is it somehow possible to control which object is being highlighted from the code?
e.g - when the user clicks a button, i want to navigate to a specific object and highlight it

**Reply by AdriSolid** [reply]
I think you are looking for `highlightedObjectIndex` prop, [check this out](https://deck.gl/docs/api-reference/core/layer#highlightedobjectindex)

**Reply by tamir-jether** [reply]
thanks!
to support something like that, i will keep the index of my current "zoomed" object in a redux state, and use that index in the highlightedObjectIndex. technically, instead of using autoHighLight, i could just change the onHover code to set the current "zoomed" index in my global state to the object my mouse is hovering, and it would work, right?  (seems like that way it wouldnt flicker like with the autoHighLight)

**Reply by AdriSolid** [reply]
Indeed should work :D
